src/Control/Carrier/FossaApiClient/Internal/FossaAPIV1.hs

```diff
@@ -366,6 +380,32 @@ getProject apiopts ProjectRevision{..} = fossaReq $ do
 
 -----
 
+getAnalyzedRevisionsEndpoint :: Url 'Https -> Url 'Https
+getAnalyzedRevisionsEndpoint baseurl = baseurl /: "api" /: "cli" /: "analyzedRevisions"
+
+-- | getAnalyzedRevisions makes a request to Core with a list of locators that we are considering scanning
+--   Core will respond with a list of locators that have already been analyzed
+getAnalyzedRevisions ::
+  (Has (Lift IO) sig m, Has Diagnostics sig m) =>
+  ApiOpts ->
+  NonEmpty VendoredDependency ->
+  m [Text]
+getAnalyzedRevisions apiOpts vDeps = fossaReq $ do
+  orgId <- organizationId <$> getOrganization apiOpts
+  (baseUrl, baseOpts) <- useApiOpts apiOpts
+  let locatorBody = constructLocatorBody orgId vDeps
+  responseBody <$> req POST (getAnalyzedRevisionsEndpoint baseUrl) (ReqBodyJson locatorBody) jsonResponse baseOpts
+  where
+    constructLocatorBody :: OrgId -> NonEmpty VendoredDependency -> GetAnalyzedRevisionsBody
+    constructLocatorBody orgId = GetAnalyzedRevisionsBody <$> NE.map (constructLocator orgId)
+      where
+        constructLocator :: OrgId -> VendoredDependency -> Text
+        constructLocator org VendoredDependency{..} = do
+          let locator = Locator{locatorFetcher = "archive", locatorProject = vendoredName, locatorRevision = vendoredVersion}
+          renderLocatorUrl org locator
```

Comment:
[required]: This function should probably be cleaned up, it's a bit messy:
- We're using `where` clauses inside of `do` notation, we should either use `let` bindings or separate functions.
- The `where` clauses are nested, with further nested inner `do` notation and `let` bindings.  I think this function can get a lot flatter.